src/herder/TxQueueLimiter.h

```diff
@@ -4,51 +4,54 @@
 // under the Apache License, Version 2.0. See the COPYING file at the root
 // of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
 
+#include "herder/SurgePricingUtils.h"
 #include "ledger/LedgerManager.h"
 #include "transactions/TransactionFrame.h"
 #include "util/UnorderedMap.h"
 #include "util/UnorderedSet.h"
 
 namespace stellar
 {
-class QueueLimiterTxMap;
+
 class TxQueueLimiter
 {
-    // size of the transaction queue, in operations
-    size_t mQueueSizeOps{0};
     // number of ledgers we can pool in memory
     uint32 const mPoolLedgerMultiplier;
     LedgerManager& mLedgerManager;
-    // minimum fee needed for a transaction
-    // stored as a pair (fee bid, nb operations)
-    std::pair<int64, uint32> mMinFeeNeeded;
+
+    std::unordered_map<TransactionFrameBasePtr, TxStackPtr> mStackForTx;
 
     // all known transactions
-    std::unique_ptr<QueueLimiterTxMap> mTxs;
-    TransactionFrameBasePtr getWorstTransaction();
+    std::unique_ptr<SurgePricingPriorityQueue> mTxs;
+
+    std::optional<uint32_t> mMaxDexOperations;
+
+    std::vector<std::pair<TxStackPtr, bool>> mEvictionCache;
```

Comment:
what I mean is that instead of this field, you could also just ask the caller to pass this around and avoid making assumptions on what the caller is doing?